SQL/database_changelog.txt

```diff
@@ -1,3 +1,24 @@
+20 May 2017, by Jordie0608
+
+Created table `round` to replace tracking of the datapoints 'round_start', 'round_end', 'server_ip', 'game_mode', 'round_end_results', 'end_error', 'end_proper', 'emergency_shuttle', 'map_name' and 'station_renames' in the `feedback` table.
+Once created this table is populated with rows from the `feedback` table.
+
+START TRANSACTION;
+CREATE TABLE `feedback`.`round` (`id` INT(11) NOT NULL AUTO_INCREMENT, `start_datetime` DATETIME NOT NULL, `end_datetime` DATETIME NULL, `server_ip` INT(10) UNSIGNED NOT NULL, `server_port` SMALLINT(5) UNSIGNED NOT NULL, `game_mode` VARCHAR(32) NULL, `game_mode_result` VARCHAR(64) NULL, `end_state` VARCHAR(64) NULL, `shuttle_name` VARCHAR(64) NULL, `map_name` VARCHAR(32) NULL, `station_name` VARCHAR(80) NULL, PRIMARY KEY (`id`));
+INSERT INTO `feedback`.`round`
+(`id`, `start_datetime`, `end_datetime`, `server_ip`, `server_port`, `game_mode`, `game_mode_result`, `end_state`, `shuttle_name`, `map_name`, `station_name`)
+SELECT DISTINCT
+@rid := round_id,
+(SELECT IFNULL((SELECT STR_TO_DATE(`details`,'%a %b %e %H:%i:%s %Y') FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "round_start"), TIMESTAMP(0))), (SELECT STR_TO_DATE(`details`,'%a %b %e %H:%i:%s %Y') FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "round_end"), (SELECT IFNULL((SELECT INET_ATON(SUBSTRING_INDEX(IF(`details` = '', '0', IF(SUBSTRING_INDEX(`details`, ':', 1) LIKE '%_._%', `details`, '0')), ':', 1)) FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "server_ip"), INET_ATON(0))), (SELECT IFNULL((SELECT IF(`details` LIKE '%:_%', CAST(SUBSTRING_INDEX(`details`, ':', -1) AS UNSIGNED), '0') FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "server_ip"), '0')), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "game_mode"), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "round_end_result"), (SELECT IFNULL((SELECT `details` FROM `feedback`.`feedback` WHERE `var_name` = "end_error"), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "end_proper"))), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "emergency_shuttle"), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "map_name"), (SELECT `details` FROM `feedback`.`feedback` WHERE `round_id` = @rid AND `var_name` = "station_renames")
```

Comment:
This seems to be erroring out because it's taking too long.

Part of that is just that the way you are doing it is excessively bad.

http://stackoverflow.com/questions/8283541/merging-multiple-rows-into-one-row-and-multiple-columns-on-mysql

Using a same table join in a group by would prevent it having to re-scan every row repeatedly for every data point for every round id (O(2,416,785^10^10000)) and instead just for every data point (O(2,416,785^10))